packages/block-editor/src/hooks/block-bindings.js

```diff
@@ -2,42 +2,164 @@
  * WordPress dependencies
  */
 import { __ } from '@wordpress/i18n';
-import { store as blocksStore } from '@wordpress/blocks';
+import { privateApis as blocksPrivateApis } from '@wordpress/blocks';
 import {
-	BaseControl,
-	PanelBody,
-	__experimentalHStack as HStack,
+	Button,
+	ButtonGroup,
 	__experimentalItemGroup as ItemGroup,
 	__experimentalItem as Item,
+	__experimentalText as Text,
+	__experimentalToolsPanel as ToolsPanel,
+	__experimentalToolsPanelItem as ToolsPanelItem,
+	privateApis as componentsPrivateApis,
 } from '@wordpress/components';
-import { useSelect } from '@wordpress/data';
+import { useSelect, useDispatch, useRegistry } from '@wordpress/data';
+import { useContext } from '@wordpress/element';
+import { useViewportMatch } from '@wordpress/compose';
 
 /**
  * Internal dependencies
  */
-import { canBindAttribute } from '../hooks/use-bindings-attributes';
+import {
+	canBindAttribute,
+	getBindableAttributes,
+} from '../hooks/use-bindings-attributes';
+import { store as blockEditorStore } from '../store';
 import { unlock } from '../lock-unlock';
 import InspectorControls from '../components/inspector-controls';
+import BlockContext from '../components/block-context';
+
+const {
+	DropdownMenuV2: DropdownMenu,
+	DropdownMenuGroupV2: DropdownMenuGroup,
+	DropdownMenuCheckboxItemV2: DropdownMenuCheckboxItem,
+	DropdownMenuItemLabelV2: DropdownMenuItemLabel,
+	DropdownMenuItemHelpTextV2: DropdownMenuItemHelpText,
+	DropdownMenuSeparatorV2: DropdownMenuSeparator,
+} = unlock( componentsPrivateApis );
+
+const useToolsPanelDropdownMenuProps = () => {
+	const isMobile = useViewportMatch( 'medium', '<' );
+	return ! isMobile
+		? {
+				popoverProps: {
+					placement: 'left-start',
+					// For non-mobile, inner sidebar width (248px) - button width (24px) - border (1px) + padding (16px) + spacing (20px)
+					offset: 259,
+				},
+		  }
+		: {};
+};
+
+function BlockBindingsPanelDropdown( {
+	fieldsList,
+	addConnection,
+	removeConnection,
+	attribute,
+	binding,
+} ) {
+	const currentKey = binding?.args?.key;
+	return (
+		<>
+			{ Object.entries( fieldsList ).map( ( [ label, fields ] ) => (
+				<>
+					<DropdownMenuGroup key={ label }>
+						{ Object.keys( fieldsList ).length > 1 && (
+							<Text
+								className="block-editor-bindings__source-label"
+								upperCase
+								variant="muted"
+							>
+								{ label }
+							</Text>
+						) }
```

Comment:
We currently don't offer dropdown menu group labels because we initially weren't sure we would want folks to use them, but it looks like you may need them here.

cc @WordPress/gutenberg-design we should probably decide if we want to allow for group labels in the component (in that case I can prepare an official dropdown menu component for that), or if that's a design pattern that should be avoided

Reply:
If a genuine need exists then I think it's fine to add. What I'm most concerned about is potential misuse leading to a proliferation of labels in menus. For instance a menu containing a single group probably shouldn't use a label/heading. Is there any way to build that constraint into the component?

Reply:
We could try to run a check at runtime, collect that sort of info, and output a warning to the console in case (only in dev/test mode). But I'd be doing that as a last resort, only if actually see folks abusing of groups and group labels.

Reply:
It would be nice to have it for this use case. Is it a problem the way it is created right now?

Reply:
It should be ok to use a custom implementation for now, but probably best to add `aria-hidden="true"`

Reply:
I added `aria-hidden` to the group heading 👍